Source/src/modules/ModuleCamera.cpp

```diff
@@ -18,39 +19,52 @@ bool Hachiko::ModuleCamera::Init()
 {
     HE_LOG("Creating main camera");
 
-    editor_camera_game_object = new GameObject("Main Camera");
-    editor_camera = static_cast<ComponentCamera*>(editor_camera_game_object->CreateComponent(Component::Type::CAMERA));
+    main_camera_game_object = new GameObject("Main Camera");
+    main_camera = static_cast<ComponentCamera*>(main_camera_game_object->CreateComponent(Component::Type::CAMERA));
 
-    editor_camera_game_object->GetTransform()->SetPosition(float3(0.0f, 8.0f, 10.0f));
-    editor_camera_game_object->GetTransform()->LookAtTarget(float3::zero);
-    editor_camera_game_object->Update();
+    main_camera->SetCameraType(ComponentCamera::CameraType::GOD);
+    main_camera_game_object->GetTransform()->SetPosition(float3(0.0f, 8.0f, 10.0f));
+    main_camera_game_object->GetTransform()->LookAtTarget(float3::zero);
+    main_camera_game_object->Update();
 
     ImGuizmo::Enable(true);
-
-    // TODO: LOOK FOR THE MAIN CAMERA
-
     return true;
 }
 
 UpdateStatus Hachiko::ModuleCamera::Update(const float delta)
 {
-    if (editor_camera_enable)
+    if (App->input->GetKey(SDL_SCANCODE_C) == KeyState::KEY_DOWN)
     {
-        Controller(delta);
+        int buffer_size = camera_buffer.size();
+        if (last_it < buffer_size)
+        {
+            main_camera = static_cast<ComponentCamera*>(camera_buffer[last_it]);
+            main_camera_game_object = camera_buffer[last_it]->GetGameObject();
+            last_it++;
+        }
+        else
+        {
+            last_it = 0;
+            main_camera = static_cast<ComponentCamera*>(camera_buffer[last_it]);
+            main_camera_game_object = camera_buffer[last_it]->GetGameObject();
+            last_it++;
+        }
```

Comment:
the whole if/else block could be replace for:

if (last_it >= buffer_size) last_it = 0;

The lines:
            main_camera = static_cast<ComponentCamera*>(camera_buffer[last_it]);
            main_camera_game_object = camera_buffer[last_it]->GetGameObject();
            last_it++;
could be outside just once as they do the same on both if / else

Reply:
This block is to be able to move through all the cameras inside the buffer when pressing "C", this is just a hardcoded feature replacing scripting, for sure we can do it better. I will take a look at what you proposed. Thank you Álvaro.